### DISCUSSION TITLE
What about split FabulousUtility into more modules for minimal dependency?

### DISCUSSION THREAD

**Discussion by ameaninglessname** [Sixze/FabulousUtility#1] [Ideas]
for example :

FabulousUtility
    - FabulousCommon : only depends on "Engine", "Core", "CoreUObject"
    - FabulousAbility: only depends on "Engine", "Core", "CoreUObject", "GameplayAbilities"
    - FabulousAI     : only depends on "Engine", "Core", "CoreUObject", "AIModule"
    ...

Users can selectively choose some modules from these rather than the only "FabulousUtility".

But I found it's more convenient for developer to choose the later approach (current one).
When keep #includes minimal(include means dependency), looks like users could still "selectively" choose dependency from other module by "include what you use".

**Comment by Sixze** [comment]
Yes, this makes sense. The plugin has grown big, so dividing it into modules will not hurt.
